### PR TITLE
Jcm sage icons

### PR DESCRIPTION
**Pull request by cmcfadzean**
## Description
We have many areas in the app where we use icons with background colors. This gives the icon an optional background color that will scale with the size of the icon to fit these needs.

## Screenshots
Screen Shot 2021-04-29 at 5 29 27 PM

## Testing in `sage-lib`
Add `card_color` to the icon component.

### CODE REVIEW

**Comment by pixelflips** [comment]
@cmcfadzean can you switch the base branch to develop, please?

File: docs/lib/sage_rails/app/views/sage_components/_sage_icon.html.erb [hunk 1]
```diff
@@ -1,12 +1,21 @@
...
-  <% if component.label.present? %>
-    aria-label="<%= component.label %>"
-  <% else %>
-    aria-hidden=true
-  <% end %>
-  <%= component.generated_html_attributes.html_safe %>
-></i>
+<% if component.card_color.present? %>
+  <div class="sage-icon-background 
+              <%= "sage-icon-background--#{component.card_color}" %>
```
Comment (QuintonJason) [inline docs/lib/sage_rails/app/views/sage_components/_sage_icon.html.erb:3]: indentation

File: docs/lib/sage_rails/app/sage_components/sage_icon.rb [hunk 2]
```diff
@@ -1,8 +1,10 @@
 class SageIcon < SageComponent
   set_attribute_schema({
     color: [:optional, NilClass, SageSchemas::COLOR_SLIDER],
+    card_color: [:optional, Set.new(["draft", "published", "info", "locked", "warning", "danger"])],
```
Comment (QuintonJason) [inline docs/lib/sage_rails/app/sage_components/sage_icon.rb:4]: Probably out of scope for this pr, but i've seen this set of colors before. I'll create a followup PR to set common color sets for additional tokens

File: docs/lib/sage_rails/app/sage_components/sage_icon.rb [hunk 3]
```diff
@@ -1,8 +1,10 @@
...
+    css_classes: [:optional, String],
```
Comment (QuintonJason) [inline docs/lib/sage_rails/app/sage_components/sage_icon.rb:5]: `css_classes` is no longer needed with a rebase on `develop` 🎉

File: docs/lib/sage_rails/app/sage_components/sage_icon.rb (hunk 2 above)
Comment (QuintonJason) [inline docs/lib/sage_rails/app/sage_components/sage_icon.rb:4]: Can you add this new prop to the `_props` file?

**Comment by QuintonJason** [comment]
Out of scope for this PR, but the `ICON_SIZES` token has duplicate `2xl` and `3xl` values.

**Review by QuintonJason — APPROVED** [review]
Looking good!

**Pull request merged**